Invalid Memory Address Segfault on 3.18.0

`Terraform 1.0.10`
`okta/okta 3.18.0`

```
Stack trace from the terraform-provider-okta_v3.18.0.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xef0779]

goroutine 775 [running]:
github.com/okta/terraform-provider-okta/okta.doesResourceExist(...)
        github.com/okta/terraform-provider-okta/okta/utils.go:245
github.com/okta/terraform-provider-okta/okta.checkIfGroupHasUsers({0x123ab68, 0xc0008b1a40}, 0xc0003b9180, {0xc00031af18, 0x14}, {0x0, 0x0, 0x0})
        github.com/okta/terraform-provider-okta/okta/resource_okta_group_memberships.go:123 +0xb9
github.com/okta/terraform-provider-okta/okta.resourceGroupMembershipsRead({0x123ab68, 0xc0008b1a40}, 0x1733c5e0eb8, {0xfec240, 0xc00024f860})
        github.com/okta/terraform-provider-okta/okta/resource_okta_group_memberships.go:72 +0xf5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0003ecfc0, {0x123aaf8, 0xc0003b9180}, 0x24, {0xfec240, 0xc00024f860})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.8.0/helper/schema/resource.go:347 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0003ecfc0, {0x123aaf8, 0xc0003b9180}, 0xc0004b12b0, {0xfec240, 0xc00024f860})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.8.0/helper/schema/resource.go:624 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00042ade0, {0x123aaf8, 0xc0003b9180}, 0xc0003b91c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.8.0/helper/schema/grpc_provider.go:576 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000498ba0, {0x123aba0, 0xc0006b0960}, 0xc0008b16e0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:298 +0x1fb
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x10b8800, 0xc000498ba0}, {0x123aba0, 0xc0006b0960}, 0xc0008b1680, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002a8a80, {0x124a8f0, 0xc0001ac180}, 0xc0008be000, 0xc00049b0e0, 0x17b6c10, 0x0)
        google.golang.org/grpc@v1.41.0/server.go:1279 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0002a8a80, {0x124a8f0, 0xc0001ac180}, 0xc0008be000, 0x0)
        google.golang.org/grpc@v1.41.0/server.go:1608 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.41.0/server.go:923 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.41.0/server.go:921 +0x294

Error: The terraform-provider-okta_v3.18.0.exe plugin crashed!
```

--- a/okta/resource_okta_group_membership.go
+++ b/okta/resource_okta_group_membership.go
@@ -101,10 +101,10 @@ func resourceGroupMembershipDelete(ctx context.Context, d *schema.ResourceData, 
 
 func checkIfUserInGroup(ctx context.Context, client *okta.Client, groupId, userId string) (bool, error) {
 	users, resp, err := client.Group.ListGroupUsers(ctx, groupId, &query.Params{Limit: defaultPaginationLimit})
-	groupExists, err := doesResourceExist(resp, err)
-	if err != nil {
+	if err := suppressErrorOn404(resp, err); err != nil {
 		return false, err
-	} else if !groupExists {
+	}
+	if len(users) == 0 {
 		return false, nil
 	}
 	for {

--- a/okta/resource_okta_group_memberships.go
+++ b/okta/resource_okta_group_memberships.go
@@ -97,10 +97,10 @@ func resourceGroupMembershipsUpdate(ctx context.Context, d *schema.ResourceData,
 	groupId := d.Get("group_id").(string)
 	client := getOktaClientFromMetadata(m)
 
-	old, new := d.GetChange("users")
+	oldUsers, newUsers := d.GetChange("users")
 
-	oldSet := old.(*schema.Set)
-	newSet := new.(*schema.Set)
+	oldSet := oldUsers.(*schema.Set)
+	newSet := newUsers.(*schema.Set)
 
 	usersToAdd := convertInterfaceArrToStringArr(newSet.Difference(oldSet).List())
 	usersToRemove := convertInterfaceArrToStringArr(oldSet.Difference(newSet).List())
@@ -120,12 +120,10 @@ func resourceGroupMembershipsUpdate(ctx context.Context, d *schema.ResourceData,
 
 func checkIfGroupHasUsers(ctx context.Context, client *okta.Client, groupId string, users []string) (bool, error) {
 	groupUsers, resp, err := client.Group.ListGroupUsers(ctx, groupId, &query.Params{Limit: defaultPaginationLimit})
-	exists, err := doesResourceExist(resp, err)
-	if err != nil {
+	if err := suppressErrorOn404(resp, err); err != nil {
 		return false, fmt.Errorf("unable to return membership for group (%s) from API", groupId)
 	}
-
-	if !exists {
+	if len(groupUsers) == 0 {
 		return false, nil
 	}
 	if resp.HasNextPage() {

--- a/okta/resource_okta_group_role.go
+++ b/okta/resource_okta_group_role.go
@@ -356,11 +356,11 @@ func findRole(ctx context.Context, d *schema.ResourceData, m interface{}) (*okta
 	}
 	groupID := d.Get("group_id").(string)
 	roles, resp, err := getOktaClientFromMetadata(m).Group.ListGroupAssignedRoles(ctx, groupID, nil)
-	exists, err := doesResourceExist(resp, err)
-	if err != nil {
+	if err := suppressErrorOn404(resp, err); err != nil {
 		return nil, fmt.Errorf("failed to list roles assigned to group %s: %v", groupID, err)
-	} else if !exists {
-		logger(m).Error("group (%s) which had these resources assigned no longer exists", groupID)
+	}
+	if len(roles) == 0 {
+		logger(m).Error("either group (%s) which had these roles assigned no longer exists or no roles were assigned", groupID)
 		return nil, nil
 	}
 	for i := range roles {

--- a/okta/resource_okta_user_group_memberships.go
+++ b/okta/resource_okta_user_group_memberships.go
@@ -116,12 +116,10 @@ func resourceUserGroupMembershipsUpdate(ctx context.Context, d *schema.ResourceD
 
 func checkIfUserHasGroups(ctx context.Context, client *okta.Client, userId string, groups []string) (bool, error) {
 	userGroups, resp, err := client.User.ListUserGroups(ctx, userId)
-	exists, err := doesResourceExist(resp, err)
-	if err != nil {
+	if err := suppressErrorOn404(resp, err); err != nil {
 		return false, fmt.Errorf("unable to return groups for user (%s) from API", userId)
 	}
-
-	if !exists {
+	if len(userGroups) == 0 {
 		return false, nil
 	}
 

--- a/okta/utils.go
+++ b/okta/utils.go
@@ -241,6 +241,9 @@ func stringPtr(s string) (ptr *string) {
 }
 
 func doesResourceExist(response *okta.Response, err error) (bool, error) {
+	if response == nil {
+		return false, err
+	}
 	// We don't want to consider a 404 an error in some cases and thus the delineation
 	if response.StatusCode == 404 {
 		return false, nil
@@ -248,7 +251,7 @@ func doesResourceExist(response *okta.Response, err error) (bool, error) {
 	if err != nil {
 		return false, responseErr(response, err)
 	}
-	return true, err
+	return true, nil
 }
 
 // Useful shortcut for suppressing errors from Okta's SDK when a resource does not exist. Usually used during deletion

